### PR TITLE
[MRG] Fix disc classifier + param grid

### PR DESCRIPTION
**Pull request by antoinedemathelin**
This PR should fix MDD and improve the param grid

### CODE REVIEW

File: solvers/deep_mdd.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Solver(DeepDASolver):
     # the cross product for each key in the dictionary.
     # All parameters 'p' defined here are available as 'self.p'.
     default_param_grid = {
-        'criterion__reg': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 1e1, 1e2, 1e3],
+        'criterion__reg': [1e-2, 1e-1, 1],
     }
```
Comment (YanisLalou) [inline solvers/deep_mdd.py:29]: For the `gamma` arg, you should define it in the param_grid like this:
```
'criterion__adapt_criterion': [
            MDDLoss(gamma=gamma, disc_criterion=torch.nn.CrossEntropyLoss())
            for gamma in [1, 2, 4]
        ],
```

Comment (YanisLalou) [reply solvers/deep_mdd.py]: Oops 🙊

Comment (antoinedemathelin) [reply solvers/deep_mdd.py]: Mais merci pour la review !

**Comment by antoinedemathelin** [comment]
Ok sorry guys but I cannot go any further with this PR.
The MDD solver works well on my computer, but I keep getting some weird import errors in the test. The last one is `NameError: name 'DeepDASolver' is not defined`. I think something went wrong in the safe import, so no package is imported...

It is too big for me guys... I tried but I failed... Please help me!

**Comment by antoinedemathelin** [comment]
Nice! Let's merge then?

**Comment by antoinecollas** [comment]
I simplified the code as much as I could to be to DANN. The training is stable on MNIST/USPS and Office31. I've got good results with the SupervisedScorer on MNIS/USPS (on-par with DANN).
I'm still running some tests on Office31 to check everything looks good.

File: solvers/deep_mdd.py [hunk 2]
```diff
@@ -25,7 +30,11 @@ class Solver(DeepDASolver):
     # the cross product for each key in the dictionary.
     # All parameters 'p' defined here are available as 'self.p'.
     default_param_grid = {
-        'criterion__reg': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 1e1, 1e2, 1e3],
+        'criterion__reg': [1e-3, 1e-2, 1e-1],
+        'criterion__adapt_criterion': [
+            MDDLoss(gamma=gamma)
+            for gamma in [2., 3.]
+        ],
```
Comment (antoinedemathelin) [inline solvers/deep_mdd.py:37]: Why not gamma in [2, 4] ? In the paper, the best param for Office31 is gamma = 4.
But ok, it should not change everything ^^

Comment (antoinedemathelin) [reply solvers/deep_mdd.py:37]: it's a matter of 0.05 points of accuracy, so forget about this

**Pull request merged**